### ISSUE TITLE
[Bugs] PFCP crashed due to `panic: assignment to entry in nil map`

### ISSUE THREAD

**Issue by pventuzelo** [free5gc/free5gc#432]
# panic: assignment to entry in nil map                                                                                                                                                                            
                                                                                                                                                                                                                   
We (@FuzzingLabs) found a 'panic: assignment to entry in nil map' panic in Free5GC v3.2.1                                                                                                                          
                                                                                                                                                                                                                  
## Expected Behavior

It should not crash.                                                                                     

## Your Environment

    c584d7b
    golang:1.14.4-stretch (docker)
    Ubuntu 20.04

## Steps to reproduce

Launch Testing program:
```
import socket
import os
UDP_IP = "10.100.200.8"
UDP_PORT = 8805
MESSAGES = []
dir_path = "crash"
MESSAGES.append(b'\x20\x05\x00\x15\x00\x00\x01\x00\x00\x3c\x00\x05\x00\x00\x00\x00\x00\x00\x60\x00\x04\xce\x0b\xd6\x5d')
MESSAGES.append(b'\x21\x32\x01\xc9\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x7a\x00\x00\x01\x00\x56\x00\x02\x00\x1f\x00\x14\x00\x01\x00\x00\x15\x00\x09\x01\x00\x00\x00\x01\xc0\xa8\x0c\xb8\x00\x16\x00\x04\x03\x65\x70\x63\x00\x7c\x00\x01\x09\x00\x1d\x00\x04\x00\x00\x00\xff\x00\x38\x00\x02\x00\x01\x00\x51\x00\x04\x00\x00\x00\x09\x00\x5f\x00\x01\x00\x00\x6c\x00\x04\x00\x00\x00\x02\x00\x6d\x00\x04\x00\x00\x00\x01\x00\x6d\x00\x04\x00\x00\x00\x02\x00\x01\x00\x48\x00\x02\x00\x16\x00\x14\x00\x01\x01\x00\x16\x00\x04\x03\x65\x70\x63\x00\x5d\x00\x05\x06\xac\x14\xe7\x01\x00\x1d\x00\x04\x00\x00\x00\xff\x00\x38\x00\x02\x00\x02\x00\x51\x00\x04\x00\x00\x00\x09\x00\x6c\x00\x04\x00\x00\x00\x03\x00\x6d\x00\x04\x00\x00\x00\x01\x00\x6d\x00\x04\x00\x00\x00\x02\x00\x01\x00\x39\x00\x02\x00\x1a\x00\x14\x00\x01\x03\x00\x15\x00\x09\x01\x00\x00\x00\x01\x00\x00\x00\x00\x00\x16\x00\x04\x03\x65\x70\x63\x00\x1d\x00\x04\x00\x00\x00\xff\x00\x38\x00\x02\x00\x03\x00\x5f\x00\x01\x00\x00\x6c\x00\x04\x00\x00\x00\x01\x00\x03\x00\x32\x00\x04\x00\x21\x00\x16\x00\x04\x03\x65\x70\x63\x00\x1e\x00\x02\x26\xff\x00\x2a\x00\x01\x00\x00\x54\x00\x0a\x01\x00\x00\x00\x00\x01\xc0\xa8\x0c\xb8\x00\x2c\x00\x01\x02\x00\x6c\x00\x04\x00\x00\x00\x01\x00\x03\x00\x1c\x00\x04\x00\x0b\x00\x1e\x00\x02\x26\xff\x00\x2a\x00\x01\x01\x00\x2c\x00\x01\x02\x00\x6c\x00\x04\x00\x00\x00\x02\x00\x03\x00\x0d\x00\x2c\x00\x01\x04\x00\x6c\x00\x04\x00\x00\x00\x03\x00\x06\x00\x1b\x00\x25\x00\x02\x01\x00\x00\x3e\x00\x01\x02\x00\x40\x00\x04\x00\x00\x00\x00\x00\x51\x00\x04\x00\x00\x00\x09\x00\x07\x00\x1b\x00\x19\x00\x01\x00\x00\x1a\x00\x0a\x00\x00\x98\x96\x80\x00\x00\x98\x96\x80\x00\x6d\x00\x04\x00\x00\x00\x01\x00\x07\x00\x12\x00\x19\x00\x01\x00\x00\x6d\x00\x04\x00\x00\x00\x02\x00\x7c\x00\x01\x09\x00\x39\x00\x0d\x02\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x3c\x00\x05\x00\x00\x00\x00\x00\x00\x71\x00\x01\x01')
MESSAGES.append(b'\x20\x05\x00\x15\x00\x00\x01\x00\x00\x3c\x00\x05\x00\x00\x00\x00\x00\x00\x60\x00\x04\xce\x0b\xd6\x5d')
sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
for message in MESSAGES:
    sock.sendto(message, (UDP_IP,UDP_PORT))
```

## Detailed behavior 

```
 goroutine 28 [running]:
runtime/debug.Stack(0xc000693838, 0x9fbe80, 0xb7dbd0)
     /usr/local/go/src/runtime/debug/stack.go:24 +0x9d
github.com/free5gc/go-upf/internal/pfcp.(\*PfcpServer).main.func1(0xc0006180d0, 0xc000373790)
     /go/src/free5gc/NFs/upf/internal/pfcp/pfcp.go:85 +0x124
panic(0x9fbe80, 0xb7dbd0)
     /usr/local/go/src/runtime/panic.go:969 +0x166
github.com/free5gc/go-upf/internal/pfcp.(\*Sess).CreateURR(0xc00061a240, 0xc000372fc0, 0x0, 0x0)
     /go/src/free5gc/NFs/upf/internal/pfcp/node.go:180 +0xb4
github.com/free5gc/go-upf/internal/pfcp.(\*PfcpServer).handleSessionEstablishmentRequest(0xc0006180d0, 0xc000246000, 0xb8dd20, 0xc000342030)
     /go/src/free5gc/NFs/upf/internal/pfcp/session.go:64 +0x536
github.com/free5gc/go-upf/internal/pfcp.(\*PfcpServer).reqDispacher(0xc0006180d0, 0xb98d40, 0xc000246000, 0xb8dd20, 0xc000342030, 0x0, 0x16)
     /go/src/free5gc/NFs/upf/internal/pfcp/dispacher.go:21 +0x1f1
github.com/free5gc/go-upf/internal/pfcp.(\*PfcpServer).main(0xc0006180d0, 0xc000373790)
     /go/src/free5gc/NFs/upf/internal/pfcp/pfcp.go:144 +0x759
created by github.com/free5gc/go-upf/internal/pfcp.(\*PfcpServer).Start
     /go/src/free5gc/NFs/upf/internal/pfcp/pfcp.go:211 +0xbd
```

**Comment by stardustlin3089** [comment]
Hi @pventuzelo. I am not able to reproduce this bug. Does PFCP still crash in your current use of free5GC? If yes, it would be helpful to provide more details on how often it occurs and how you builds free5GC.

**Comment by pventuzelo** [comment]
Hey,

Yes it's still crashing with the following errors:
```
free5gc_upf      | 2023-06-02T08:12:52Z [INFO][UPF][Pfcp][upf.free5gc.org:8805] handleAssociationSetupRequest
free5gc_upf      | 2023-06-02T08:12:52Z [INFO][UPF][Pfcp][upf.free5gc.org:8805][rNodeID:0.0.0.0] New node
free5gc_upf      | 2023-06-02T08:12:52Z [INFO][UPF][Pfcp][upf.free5gc.org:8805] handleSessionEstablishmentRequest
free5gc_upf      | 2023-06-02T08:12:52Z [INFO][UPF][Pfcp][upf.free5gc.org:8805][rNodeID:0.0.0.0][SEID:L(0x1),R(0x1)] New session
free5gc_upf      | 2023-06-02T08:12:52Z [FATA][UPF][Pfcp][upf.free5gc.org:8805] panic: assignment to entry in nil map
free5gc_upf      | goroutine 52 [running]:
free5gc_upf      | runtime/debug.Stack(0xc000093838, 0x9fbe80, 0xb7dbd0)
free5gc_upf      | 	/usr/local/go/src/runtime/debug/stack.go:24 +0x9d
free5gc_upf      | github.com/free5gc/go-upf/internal/pfcp.(*PfcpServer).main.func1(0xc000312000, 0xc0004fc350)
free5gc_upf      | 	/go/src/free5gc/NFs/upf/internal/pfcp/pfcp.go:85 +0x124
free5gc_upf      | panic(0x9fbe80, 0xb7dbd0)
free5gc_upf      | 	/usr/local/go/src/runtime/panic.go:969 +0x166
free5gc_upf      | github.com/free5gc/go-upf/internal/pfcp.(*Sess).CreateURR(0xc0000a04e0, 0xc0000a33c0, 0x0, 0x0)
free5gc_upf      | 	/go/src/free5gc/NFs/upf/internal/pfcp/node.go:180 +0xb4
free5gc_upf      | github.com/free5gc/go-upf/internal/pfcp.(*PfcpServer).handleSessionEstablishmentRequest(0xc000312000, 0xc00013c000, 0xb8dd20, 0xc000530a50)
free5gc_upf      | 	/go/src/free5gc/NFs/upf/internal/pfcp/session.go:64 +0x536
free5gc_upf      | github.com/free5gc/go-upf/internal/pfcp.(*PfcpServer).reqDispacher(0xc000312000, 0xb98d40, 0xc00013c000, 0xb8dd20, 0xc000530a50, 0x0, 0x16)
free5gc_upf      | 	/go/src/free5gc/NFs/upf/internal/pfcp/dispacher.go:21 +0x1f1
free5gc_upf      | github.com/free5gc/go-upf/internal/pfcp.(*PfcpServer).main(0xc000312000, 0xc0004fc350)
free5gc_upf      | 	/go/src/free5gc/NFs/upf/internal/pfcp/pfcp.go:144 +0x759
free5gc_upf      | created by github.com/free5gc/go-upf/internal/pfcp.(*PfcpServer).Start
free5gc_upf      | 	/go/src/free5gc/NFs/upf/internal/pfcp/pfcp.go:211 +0xbd
free5gc_upf exited with code 1
```

we are using the version: free5gc/upf:v3.2.1

**Comment by zyddnys** [comment]
fixed by changing `NewSess` function of `node.go:377` to
```go
s := &Sess{
		RemoteID: rSeid,
		PDRIDs:   make(map[uint16]struct{}),
		FARIDs:   make(map[uint32]struct{}),
		QERIDs:   make(map[uint32]struct{}),
		URRIDs:   make(map[uint32]struct{}),
		q:        make(map[uint16]chan []byte),
		qlen:     qlen,
	}
```

**Comment by ianchen0119** [comment]
Hi @pventuzelo 

Thanks for your report.
This issue has been fixed in free5GC v3.3.0.
